libretro-common/include/libretro.h

```diff
@@ -976,6 +976,19 @@ struct retro_hw_render_context_negotiation_interface
                                             * so it will be used after SET_HW_RENDER, but before the context_reset callback.
                                             */
 
+struct retro_resource
+{
+   const char *rel_path;
+   const char *base_path;
```

Comment:
It's unlikely that the frontend path will be longer than size_t because it's just a path prefix, not a possibly-very-deep leaf

If we're going to do tricky buffer management, I wouldn't want to expose the underlying buffer. This should be abstracted away, but now we're deep in abstractions for something that can be solved by the core managing the buffer.